engine/buildkit/ref.go

```diff
@@ -197,6 +197,13 @@ func (r *ref) Result(ctx context.Context) (bksolver.CachedResult, error) {
 	return res, nil
 }
 
+func (r *ref) Provenance() interface{} {
```

Comment:
First, super cool this is so easy, so many interesting possibilities to explore in general w/ provenance 😎

second, as a nit, rather than returning `interface{}` here, could we do something like:
```go
func (r *ref) ProvenanceCapture() (*provenance.Capture, bool)
```

just to avoid the need for the separate `resolveProvenance` util (and probably other utils as we want to handle more types in the future)?

Reply:
Yeah, we only ever assign `*provenance.Capture` to this. I can't *quite* remember why it's as generic as it is, but we should be good for now to layer it as you suggest.

If it ever changes upstream, we can tidy it up here.